izpack-installer/src/main/java/com/izforge/izpack/installer/unpacker/CompressedFileUnpacker.java

```diff
@@ -72,17 +72,21 @@ public void unpack(PackFile file, InputStream packInputStream, File target)
             throws IOException, InstallerException
     {
         File tmpfile = File.createTempFile("izpack-uncompress", null, FileUtils.getTempDirectory());
+        final long fileSize = file.size();
+        final long fileLength = file.length();
+        final long backReferenceFileLength = file.isBackReference() ? file.getLinkedPackFile().size() : fileSize;
+
         OutputStream fo = null;
         InputStream finalStream = null;
 
         try
         {
             fo = IOUtils.buffer(FileUtils.openOutputStream(tmpfile));
-            final long bytesUnpacked = IOUtils.copyLarge(packInputStream, fo, 0, file.size());
+            final long bytesUnpacked = IOUtils.copyLarge(packInputStream, fo, 0, fileSize);
```

Comment:
@reinhapa  Why is this not `backReferenceFileLength` instead of `fileSize`?

Reply:
We have cases where the `length` and `size` on the `PackFile` instance where the same but the `size` on the back referenced `PackFile` was different. I was never able to get the correct result by only using one property. It seems to me that this check should use an SHA-xx has at the end anyway and this should only be an intermediate fix though....

Reply:
I might have formulated my question unclear: 
On line 77 you get the `backReferenceFileLength` depending on the size of the `linkedPackFile` or otherwise use the `fileSize`.
On line 85 you copy the content always with `fileSize` to get the `bytesUnpacked` and not with `backReferenceFileLength` (instead of `fileSize`). I was just wondering how on line 89 the term `bytesUnpacked == backReferenceFileLength` can become true.
If that is correct then I'm fine with the change.

Reply:
This is what we also did not found out completely, only that we got the following combinations:
- `bytesUnpacked` equal to `fileSize` that represented the non inflated size
- `bytesUnpacked` equal to `fileLength` that represented the inflated size
- `bytesUnpacked` equal to `backRefenceFileLength` the represented the non inflated size whereas `fileSize` and `fileLength` that represented both the inflated size